docs/pages/kubernetes-access/discovery/aws.mdx

```diff
@@ -0,0 +1,232 @@
+Kubernetes RBAC group such as `system:masters`, but Teleport recommends setting 
+up a separate Kubernetes RBAC group with the minimum required permissions.
+
+<Tabs>
+  <TabItem label="Minimum RBAC permissions (recommended)" >
+
+Connect to your target cluster with your credentials and create the following 
+resources using `kubectl`.
+
+### ClusterRole
+
+Create the `ClusterRole` RBAC definition with the required permissions for Teleport 
+Kubernetes Service to forward requests to the cluster.
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: teleport
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - users
+  - groups
+  - serviceaccounts
+  verbs:
+  - impersonate
+- apiGroups:
+  - ""
+  resources:
+  - pods
+  verbs:
+  - get
+- apiGroups:
+  - "authorization.k8s.io"
+  resources:
+  - selfsubjectaccessreviews
+  - selfsubjectrulesreviews
+  verbs:
+  - create
+```
+
+### ClusterRoleBinding
+
+Link the previously created `ClusterRole` into a `teleport` RBAC group.
+
+```yaml
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: teleport
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: teleport
+subjects:
+- kind: Group
+  name: teleport
+  apiGroup: rbac.authorization.k8s.io
+```
+
+### IAM mapping
+
+Finally, edit the `configmap/aws-auth` in the `kube-system` namespace and append 
+the following to `mapRoles`. Replace `{teleport_aws_iam_role}` with the 
+appropriate IAM role that Teleport Kubernetes Service will use. This step will 
+link the Teleport IAM role into the Kubernetes RBAC group `teleport`.
+
+```yaml
+apiVersion: v1
+data:
+  mapRoles: |
+    - groups:
+      - teleport
+      rolearn: {teleport_aws_iam_role} # e.g. arn:aws:iam::222222222222:role/teleport-role
+      username: teleport
+```
+
+  </TabItem>
+  <TabItem label="system:masters permissions" >
+    <Notice type="warning">
+Granting the `system:masters` group to the IAM role associated with the Teleport 
+service means granting administrator-level permissions on the Kubernetes cluster.  **Do not use this method in production.**
+  </Notice>
+
+To associate the Teleport IAM role with the `system:masters` RBAC group, 
+edit the `configmap/aws-auth` in the `kube-system` namespace and append 
+the following to `mapRoles`. 
+
+```yaml
+apiVersion: v1
+data:
+  mapRoles: |
+  ...
+    - groups:
+      - system:masters
+      rolearn: {teleport_aws_iam_role} # e.g. arn:aws:iam::222222222222:role/teleport-role
+      username: teleport
+```
+
+Please replace `{teleport_aws_iam_role}` with the appropriate IAM role that 
+Teleport Kubernetes Service is using.
+
+  </TabItem>
+</Tabs>
+
+At this point, the Teleport IAM role already has the minimum permissions 
+to forward requests to the cluster. 
+
+
+## Step 3/3. Configure Teleport to discover EKS clusters
+
+Teleport EKS Auto-Discovery requires a valid auth token for the Discovery and 
```

Comment:
```suggestion
Teleport EKS Auto-Discovery requires a valid Teleport auth token for the Discovery and 
```

otherwise might not be clear, is it Teleport token or AWS token or something else.